youtube_dl/extractor/iai.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class IaiIE(InfoExtractor):
+    IE_NAME = 'IAI'
+    _VALID_URL = r'https://iai.tv/video/(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://iai.tv/video/darkness-authority-and-dreams',
+        'md5': '633f3175a9be21a10bc0583acc5d06bb',
+        'info_dict': {
+            'id': 'darkness-authority-and-dreams',
+            'ext': 'mp4',
+            'title': 'Darkness, Authority and Dreams',
+            'description': 'md5:1dbbaabf0609bcb45220b2b7d9ad063f',
+            'upload_date': '20190404',
+            'timestamp': 1554379576,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        data = self._search_regex(r'\bwindow\.JS_VARS\s*=(.*)', webpage, 'JS_VARS')
+        data = self._parse_json(data, video_id)
+        formats = []
+        for widget in data['Widgets']['widgets']:
+            if widget.get('Type') != 'VideoPlayerWidget':
+                continue
+            sprops = widget['Sprops']
+            for format_id in 'sd', 'hd':
+                quality = 1 if format_id == 'hd' else 0
+                key = ('HighRes' if quality > 0 else '') + 'StreamingUrl'
```

Comment:
34-35 can be easily moved into `for`.

Reply:
Yes, but why?

Reply:
DRY.

Reply:
I don't see anything wrong with this code.